jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/pinning/ConnectionPinningManager.kt

```diff
@@ -93,32 +90,27 @@ class DefaultConnectionPinningManager :
         ApplicationManager.getApplication().messageBus.syncPublisher(ConnectionPinningManagerListener.TOPIC).pinnedConnectionChanged(feature, newConnection)
     }
 
-    override fun maybePinFeatures(oldConnection: ToolkitConnection?, newConnection: ToolkitConnection, features: List<FeatureWithPinnedConnection>) {
+    override fun pinFeatures(oldConnection: ToolkitConnection?, newConnection: ToolkitConnection, features: List<FeatureWithPinnedConnection>) {
         val featuresString = if (features.size == 1) {
             features.first().featureName
         } else {
             "${features.dropLast(1).joinToString(",") { it.featureName }} and ${features.last().featureName}"
         }
 
-        var connectionToPin = if (oldConnection is AwsBearerTokenConnection) oldConnection else newConnection
-        if (showDialogIfNeeded(oldConnection, newConnection, featuresString)) {
-            features.forEach {
-                setPinnedConnection(it, connectionToPin)
-            }
-            notifyInfo(message("credentials.switch.notification.title", featuresString, connectionToPin.label))
+        val connectionToPin = if (oldConnection is AwsBearerTokenConnection) oldConnection else newConnection
```

Comment:
In the case where both Identity Center and Builder ID can be used and we are switching connections from Builder ID to Identity Center, should we still pin the connection?

Reply:
right, we keep the connection they started with